extensions/schema-compare/src/models/schemaCompareOptionsModel.ts

```diff
@@ -9,217 +9,33 @@ import { isNullOrUndefined } from 'util';
-			, this.deploymentOptions.allowDropBlockingAssemblies.displayName
-			, this.deploymentOptions.dropConstraintsNotInSource.displayName
-			, this.deploymentOptions.dropDmlTriggersNotInSource.displayName
-			, this.deploymentOptions.dropExtendedPropertiesNotInSource.displayName
-			, this.deploymentOptions.dropIndexesNotInSource.displayName
-			, this.deploymentOptions.ignoreFileAndLogFilePath.displayName
-			, this.deploymentOptions.ignoreExtendedProperties.displayName
-			, this.deploymentOptions.ignoreDmlTriggerState.displayName
-			, this.deploymentOptions.ignoreDmlTriggerOrder.displayName
-			, this.deploymentOptions.ignoreDefaultSchema.displayName
-			, this.deploymentOptions.ignoreDdlTriggerState.displayName
-			, this.deploymentOptions.ignoreDdlTriggerOrder.displayName
-			, this.deploymentOptions.ignoreCryptographicProviderFilePath.displayName
-			, this.deploymentOptions.verifyDeployment.displayName
-			, this.deploymentOptions.ignoreComments.displayName
-			, this.deploymentOptions.ignoreColumnCollation.displayName
-			, this.deploymentOptions.ignoreAuthorizer.displayName
-			, this.deploymentOptions.ignoreAnsiNulls.displayName
-			, this.deploymentOptions.generateSmartDefaults.displayName
-			, this.deploymentOptions.dropStatisticsNotInSource.displayName
-			, this.deploymentOptions.dropRoleMembersNotInSource.displayName
-			, this.deploymentOptions.dropPermissionsNotInSource.displayName
-			, this.deploymentOptions.dropObjectsNotInSource.displayName
-			, this.deploymentOptions.ignoreColumnOrder.displayName
-			, this.deploymentOptions.ignoreTablePartitionOptions.displayName
-			, this.deploymentOptions.doNotEvaluateSqlCmdVariables.displayName
-			, this.deploymentOptions.disableParallelismForEnablingIndexes.displayName
-			, this.deploymentOptions.disableIndexesForDataPhase.displayName
-			, this.deploymentOptions.restoreSequenceCurrentValue.displayName
-			, this.deploymentOptions.rebuildIndexesOfflineForDataPhase.displayName
-			, this.deploymentOptions.preserveIdentityLastValues.displayName
-			, this.deploymentOptions.isAlwaysEncryptedParameterizationEnabled.displayName
-			, this.deploymentOptions.allowExternalLibraryPaths.displayName
-			, this.deploymentOptions.allowExternalLanguagePaths.displayName
-			, this.deploymentOptions.hashObjectNamesInLogs.displayName
-			, this.deploymentOptions.doNotDropWorkloadClassifiers.displayName
-			, this.deploymentOptions.ignoreWorkloadClassifiers.displayName
-			, this.deploymentOptions.ignoreDatabaseWorkloadGroups.displayName
-			, this.deploymentOptions.doNotDropDatabaseWorkloadGroups.displayName].sort();
+	public UpdateOptionsMapTable() {
+		this.optionsMapTable = this.deploymentOptions.optionsMapTable;
 	}
 
 	public getOptionsData(): string[][] {
 		let data = [];
-		this.optionsLookup = {};
+		this.optionsLookup = new Map<string, boolean>();
 		this.optionsLabels.forEach(l => {
 			let checked: boolean = this.getSchemaCompareOptionUtil(l);
 			data.push([checked, l]);
```

Comment:
in the same function in the sql database projects PR, there's a check for if `checked` is undefined. Is that needed here too?

Reply:
No need here, as there the optionsmodel could be undefined so the checked variable  is expected to be undefined too. here the getSchemaCompareOptionUtil never returns undefined values as SQL DB model method.